examples/closure-app/closure-common/src/ClosureControlEndpoint.cpp

```diff
@@ -159,12 +159,73 @@ CHIP_ERROR ClosureControlEndpoint::Init()
 
 void ClosureControlEndpoint::OnStopCalibrateActionComplete()
 {
-    // This function should handle closure control state updation after stopping of calibration Action.
+    VerifyOrReturn(mLogic.SetMainState(MainStateEnum::kStopped) == CHIP_NO_ERROR,
+                   ChipLogError(AppServer, "Failed to set main state in OnStopCalibrateActionComplete"));
+
+    // After stopping calibration, the overall and target state is explicitly nulled to indicate an unknown state,
+    VerifyOrReturn(mLogic.SetOverallCurrentState(DataModel::NullNullable) == CHIP_NO_ERROR,
+                   ChipLogError(AppServer, "Failed to set overall state to null in OnStopCalibrateActionComplete"));
+    VerifyOrReturn(mLogic.SetOverallTargetState(DataModel::NullNullable) == CHIP_NO_ERROR,
+                   ChipLogError(AppServer, "Failed to set overall target to null in OnStopCalibrateActionComplete"));
+    VerifyOrReturn(mLogic.SetCountdownTimeFromDelegate(0) == CHIP_NO_ERROR,
+                   ChipLogError(AppServer, "Failed to set countdown time to 0 in OnStopCalibrateActionComplete"));
+    VerifyOrReturn(mLogic.GenerateMovementCompletedEvent() == CHIP_NO_ERROR,
+                   ChipLogError(AppServer, "Failed to generate movement completed event in OnStopCalibrateActionComplete"));
 }
 
 void ClosureControlEndpoint::OnStopMotionActionComplete()
 {
-    // This function should handle closure control state updation after stopping of Motion Action.
+    VerifyOrReturn(mLogic.SetMainState(MainStateEnum::kStopped) == CHIP_NO_ERROR,
+                   ChipLogError(AppServer, "Failed to set main state in OnStopMotionActionComplete"));
+
+    // Set the OverallState position to PartiallyOpened as motion has been stopped
+    // and the closure is not fully closed or fully opened.
+    auto position = MakeOptional(DataModel::MakeNullable(CurrentPositionEnum::kPartiallyOpened));
```

Comment:
The problem with this assumption is that the device could be in a fully closed or fully open position when the stop command is received. There is no garantee that the device is in fact partially opened.

For example, device is fully closed and waiting for motion thus is has not moved and receives a stop command. Is it fully closed or partially opened?